### PR TITLE
Partition mount

### PR DESCRIPTION
**Pull request by mathematicw**
Another menu item on the left side of the documentation page.
A detailed guide to partitioning, creating file systems (XFS, EXT4, VFAT and BTRFS) and mounting, with links to recognized manuals. There is useful information on computer science. Of course, this tutorial could be updated and improved. But overall, I can see how this kind of material could be useful, because the idea is to get all the information you need to run a great farm in one place. 
In order to do this brilliantly, it might take some time to properly assess what information is unnecessary and what information is actually useful for most future community members.
I suppose it makes sense to add RAID in the near future, and something will probably turn out to be unnecessary :)

### CODE REVIEW

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+MBR (Master Boot Record) and GPT (GUID Partition Table) are partitioning schemes that define how data is organized on a storage device.
+
+    The main partition table in MBR partitioning scheme starts in the very first sector of the storage device (the 0th sector). This is also known as the Master Boot Record sector. This sector is a 512-byte region that contains both the partition table and the boot code. The partition table occupies the first 446 bytes, while the remaining 66 bytes are reserved for the boot code. The boot code is executed by the computer's BIOS during the boot process and is responsible for locating and loading the operating system's boot loader, which is stored in the active partition's boot sector. There are recommendations for [partition codes](https://en.wikipedia.org/wiki/GUID_Partition_Table): 0x07 for NTFS, 0x83 for Linux, 0x0B for FAT32, and for file systems for boot partition in MBR scheme: for Windows, the boot disk is usually formatted with the NTFS file system, for Linux, it might be formatted with ext4. The recommended size of this partition is between 100 and 200 MiB.
```
[the remark this answers is not in the document]
Comment (EmilFattakhov) [reply versioned_docs/version-latest/protocol/Partition-mount.md:28]: Formatting for the section looks incorrect and shows up a one very long string on a web version.

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    nvme list       
+    nvme --smart-log /dev/ng0n1
+    nvme --smart-log /dev/ng1n1
+```
+
+## Partitioning
+
+For partition manipulation, the most popular utilities are:
+
+#### [parted](https://www.gnu.org/software/parted/)
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:52]:
```suggestion
#### [Parted](https://www.gnu.org/software/parted/)
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+##### Works with both GPT and MBR partition tables. It can resize partitions, but it does not resize the file systems within them. To resize file systems, you may need to use utilities like e2fsprogs separately. Note that parted may have some alignment issues with GPT partitions.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:53]: Doesn't need to be a header, please remove ####

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+#### [e2fsprogs](http://e2fsprogs.sourceforge.net/)
+##### This utility is specifically designed for working with ext2 and ext3 file systems.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:56]: Doesn't need to be a header

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+
+#### [Blocks to Gigabytes Calculator](https://www.unitconverters.net/data-storage/block-to-gigabyte.htm)
+##### This online tool can help you convert between different storage measurement units.
+
+#### [gdisk](https://www.rodsbooks.com/gdisk/) and [its non-interactive version, sgdisk](https://www.rodsbooks.com/gdisk/sgdisk.html)
+##### These tools are primarily for GPT partitions. They are useful for tasks such as reordering partitions, aligning to the end, and labeling. However, they cannot resize partitions. 
+sgdisk can be used as automated script for partition tasks on multiple disks or partitions.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:63]:
```suggestion
Sgdisk can be used as an automated script for partition tasks on multiple disks or partitions.
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+    
+    sgdisk -z /dev/nvme1n1 -n 1::+512M -t 1:EF00 -c 1:boot -v /dev/nvme1n1 --align-end -n 2::-4076M -t 2:8300 -c 2:main -v /dev/nvme1n1 -n 3::+2038M -t 3:8300 -c 3:j1 -v /dev/nvme1n1 -n 4::+2038M -t 4:8300 -c 4:j2 -v /dev/nvme1n1 -p /dev/nvme1n1
+    
+ -v	checks for a variety of problems, such as incorrect CRCs and mismatched main and backup data. But does not automatically correct most problems; -c	change-name=partnum:name; -t partition_number:hexcode; -p prints info about this device; -z	wipes tables and all partitions; -I, --align-end aligns the end of the partition to a 2048-sector boundary to the end of disk.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:69]: Let's make `-v` a code snippet for readability

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+
+ *unlike gdisk sgdisk does not allign partitions in 2048 boundaries to the end of disk by default, so need to use --align-end option before the command that creates a partition.*
+
+
+### Resize partition 
+During partition resizing, you need to make the size of the hive system smaller than (or equal to) the partition. First reduce it, resize the partition, and then adjust the fs to the size of the partition without explicitly specifying the size in the resize2fs tool. When u resize partition with parted: copy old partition's GUID, delete old partition, create new one matching starting sector! (otherwise data will be lost), set old GUID to the new one. 
+
+Useful tools are:
+
+#### [calculator](https://www.unitconverters.net/data-storage/block-to-gigabyte.htm) blocks to gigabytes
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:79]:
```suggestion
#### [Calculator](https://www.unitconverters.net/data-storage/block-to-gigabyte.htm) blocks to gigabytes
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+mkfs.vfat -c -n "boot" -F 32 -v /dev/nvme1n1p1  # c -check for errors; n-label; F -FAT size, v-verbose
+```
+
+## EXT(3,4)
+#### creating EXT4 filesystem with setting it's name (farmer-0):
+```console
+mkfs.ext4 -L "farmer-0" /dev/nvme0n1p1  # -L - label
+```
+#### Creating small EXT4 partition which will be used as external journal:
+###### note: external-journal have same block size as it's main filesystem (normally, it is so by default)
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:97]: You can use 
```
:::note 
your note
:::
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 9]
```diff
@@ -0,0 +1,411 @@
...
+* When creating XFS file system, you should specify an external log device using the `-l logdev=device` option with the `mkfs.xfs` command. 
+If you do not specify a log device duiring creating the file system, you will not be able to add one later
+* You don't need to mount the journal separately! - Just the main partition mount is enough. The log will be hooked up automatically.
+
+Some interesting functions of XFS filsystem:
+    
+    -l log_section_options (These options specify the location, size, and other parameters of the log section of the filesystem. The valid log_section_options are: `logdev=device`. This is used to specify that the log section should reside on the device separate from the data section.
+
+    nobarrier (For ex. : `mount -o nobarrier /dev/device /mount/point`) For devices with non-volatile, battery-backed write caches and those with write-caching disabled, you can safely disable write barriers at mount time using the -o nobarrier option for mount. The role of write barriers in maintaining data integrity during power failures when dealing with storage devices that have extra caches.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:146]: Wrong formatting, this is shown as a code block, while a lot of it is a plaint text. This is the case for the next 3 sections.

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 10]
```diff
@@ -0,0 +1,411 @@
...
+Btrfs aims to provide improved scalability and reliability compared to Ext4, addressing weaknesses in existing Linux file systems. It focuses on fault tolerance, repair capabilities, and straightforward administration.
+Ext4 is a journaling file system, which can be slow in checking large hard disks, particularly as disk sizes grow. Btrfs handles this more efficiently.
+Ext4 has limits, supporting up to 1 exbibyte for partitions and 16 tebibytes for file sizes. Btrfs, however, supports up to 16 exbibytes for partitions and files, making it significantly more scalable and is essential for enterprise environments and cloud storage. Btrfs isn't currently stable due to its ongoing development. In some sources it's considered experimental.
+
+
+Some interesting features
+
+    Unlike some other file systems that support external journals, such as XFS and ext4, Btrfs does not support external journaling. The journaling in Btrfs is integrated) into its COW mechanism and is not a separate feature that can be moved to another device.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:186]: Formatting issues with this section as well.

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 11]
```diff
@@ -0,0 +1,411 @@
...
+
+[Manual btrfs](https://www.man7.org/linux/man-pages/man8/btrfs.8.html)
+[BTRFS device](https://www.man7.org/linux/man-pages/man8/btrfs-device.8.html)
+[BTRFS filesystem](https://www.man7.org/linux/man-pages/man8/btrfs-filesystem.8.html)
+
+
+## Mount
+
+`/mnt` directory is traditionally used for temporary mount points. For permanent mount points it's common to create a dedicated directory for the mount point.
+ChatGPT
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:231]: Looks unrelated

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 12]
```diff
@@ -0,0 +1,411 @@
...
+
+All systems, files, and devices can be mounted and unmounted as needed to adjust their properties (refer to the "Some useful mounting options" section below).
+
+## mount commands
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:235]:
```suggestion
## Mount commands
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 13]
```diff
@@ -0,0 +1,411 @@
...
+
+#### This command usually given in a bootscript causes all filesystems mentioned in fstab to be mounted, exept with -noauto flag:
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:237]: This one and similar below don't need to be a header

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 14]
```diff
@@ -0,0 +1,411 @@
...
+
+
+- To check if COW is enabled on a file or directory, you can use the `lsattr` command. For example:
+  ```console
+  lsattr /path/to/file
+  lsattr -d /path/to/directory
+  ```
+
+
+### chroot
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:277]:
```suggestion
### Chroot
```

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 15]
```diff
@@ -0,0 +1,411 @@
...
+```console
+mount -a
+```
+
+
+##	Addon:  some useful mounting options
+
+### for all file systems
+
+async
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:338]: Let's put all code examples below as code snippets

File: versioned_docs/version-latest/protocol/Partition-mount.md [hunk 16]
```diff
@@ -0,0 +1,411 @@
...
+rw
+ Mount the filesystem read-write
+ro
+ Mount the filesystem read-only. 
+### for EXT4: 
+max_batch_time=usec
+    Maximum amount of time ext4 should wait for additional filesystem operations to be batch together with a synchronous write operation. Since a synchronous write operation is going to force a commit and then a wait for the I/O complete, it doesn't cost much, and can be a huge throughput win, we wait for a small amount of time to see if any other transactions can piggyback on the synchronous write. The algorithm used is designed to automatically tune for the speed of the disk, by measuring the amount of time (on average) that it takes to finish committing a transaction. Call this time the "commit time". If the time that the transactoin has been running is less than the commit time, ext4 will try sleeping for the commit time to see if other operations will join the transaction. The commit time is capped by the max_batch_time, which defaults to 15000us (15ms). This optimization can be turned off entirely by setting max_batch_time to 0.
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/Partition-mount.md:364]: Formatting

File: versioned_docs/version-latest/protocol/pulsar.mdx [hunk 17]
```diff
@@ -33,14 +33,12 @@ Pulsar requires commodity hardware specs to operate. At a **minimum**, it is rec
 
 | Hardware | Specs            |
 | -------- | ---------------- |
-| CPU      | 4 Core+          |
-| RAM      | 8GB+             |
-| SWAP     | 4GB              |
-| Storage  | 100GB SSD        |
+| CPU      | 4 Core           |
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/pulsar.mdx:36]: Looks like this section shouldn't be changed

File: versioned_docs/version-latest/protocol/security.md [hunk 18]
```diff
@@ -1,86 +1,266 @@
 ---
-title: Safety and Security
-sidebar_position: 6
-description: Security and Scam Prevention
+title: General and server security guidelines
```
Comment (EmilFattakhov) [inline versioned_docs/version-latest/protocol/security.md:2]: These changes are reflected in your other PR, let's keep the improvements separate and remove the changes from this PR, please :)

**Review by EmilFattakhov — CHANGES_REQUESTED** [review]
Thank you for your contribution, a lot of useful information here. 
Requesting some improvements mostly with formatting, display and wording. 
Let's also remove all the changes that are already reflected in your other PRs, e.g. we can remove the whole `security.md` file

File: versioned_docs/version-latest/protocol/Partition-mount.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathematicw) [reply versioned_docs/version-latest/protocol/Partition-mount.md:28]: This is done to make it easy to read and compact at the same time. I have specially placed this text in a block with the "Toggle word wrap" function. There is a functionality that allows people to read this text comfortably, not as one very long line, but as normal text block.

[the remark this answers is not in the document]
Comment (mathematicw) [reply versioned_docs/version-latest/protocol/Partition-mount.md:28]: But it's probably worth putting the whole text on the page. As it is not so large

[the remark this answers is not in the document]
Comment (mathematicw) [reply versioned_docs/version-latest/protocol/Partition-mount.md:28]: > This section is duplicated below with slightly different information and numbers.

Are you suggesting to combine these two quite similar sections (the first which is about MBR and the second which is about GPT) into one text block and thus reduce duplicates?

**Comment by mathematicw** [comment]
> Thank you for your contribution, a lot of useful information here. Requesting some improvements mostly with formatting, display and wording. Let's also remove all the changes that are already reflected in your other PRs, e.g. we can remove the whole `security.md` file

I'm sorry, I didn't catch your point. Did we speak about deleting whole security.md file?

**Comment by mathematicw** [comment]
> Thank you for your contribution, a lot of useful information here. Requesting some improvements mostly with formatting, display and wording. Let's also remove all the changes that are already reflected in your other PRs, e.g. we can remove the whole `security.md` file

Ok

**Comment by mathematicw** [comment]
Yeah, I can see there's a lot of work to be done here. It's gonna take me a couple days to get it right.

File: versioned_docs/version-latest/protocol/Partition-mount.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (EmilFattakhov) [reply versioned_docs/version-latest/protocol/Partition-mount.md:28]: Yes, I think we can combine the information from both of them to one section.

**Comment by EmilFattakhov** [comment]
> Yeah, I can see there's a lot of work to be done here. It's gonna take me a couple days to get it right.

**Comment by mathematicw** [comment]
Do we need this 'partition mounting' article?

**Comment by mathematicw** [comment]
I deleted this branch. Hope this is correct action. 
And created new PR 383 with only one file : file-systems.md

**Pull request closed without merging**